src/wasm/wasm-binary.cpp

```diff
@@ -2540,7 +2549,7 @@ void WasmBinaryBuilder::readTableElements() {
       throwError("Table index out of range.");
     }
 
-    if (headerFormat == 0x2) {
+    if (flags == 0x2) {
```

Comment:
It would be good to reformulate this condition in terms of `isInactive` and friends.

Reply:
I merged the my enum into `SegmentFlag` as they have the exact same meaning for both table and memory segments. The only difference is the third bit which is exclusive to table segments at the moment. Also fixed this.